backend/src/ml_space_lambda/data_access_objects/dataset.py

```diff
@@ -38,6 +38,7 @@ def __init__(
         created_by: str,
         created_at: Optional[float] = None,
         last_updated_at: Optional[float] = None,
+        groups: Optional[List] = [],
```

Comment:
Do we want this as `Optional` if we're always defaulting it to `[]`? In what scenario do we expect it to be `None` instead of at worst an empty list?

Reply:
Fair point, I'll just change it to defaulting to an empty list and remove the Optional bit